### DISCUSSION TITLE
Custom Authentication

### DISCUSSION THREAD

**Discussion by kchojhu** [supabase/supabase#5264] [General]
Just discovered this gem but I think it needs custom authentication.  I have some use case where the authentication is not OAUTH2 compatible and I have no control over that mechanism.  It would be great if there's a way to create custom JS/TS script that implements specific interfaces (ex.  implements AuthorizationService) and drop it into some locations.  This way it can handle just about any use case to deal with authentication.  In my case, I like to use client's certificate to do the validation through HTTPS.

**Comment by kiwicopple** [comment]
Hey @kchojhu - thanks for dropping in with some ideas.

It's not entirely clear what the steps are here - it sounds like you need some advice on how to implement something custom? I'll convert this to a GitHub Discussion, and feel free to share more details

**Comment by kchojhu** [comment]
Thanks @kiwicopple .  Maybe it does support client certificate based authorization.  Just to be sure we're on the same page, here's the info on client certificate https://www.ssl.com/how-to/configuring-client-authentication-certificates-in-web-browsers/

Hope someone can help.

**Comment by kchojhu** [comment]
Looks like someone already posted here https://github.com/supabase/supabase/discussions/417 with a solution.  I hope Supabase would have official support for custom auth.

**Comment by CanRau** [comment]
Also found this article [Using Supabase RLS with a custom auth provider](https://medium.com/@gracew/using-supabase-rls-with-a-custom-auth-provider-b31564172d5d) which looks promising, trying out now 🧑‍🔧

Also if anyone, like me, is doing this from Cloudflare Pages, this might come in handy [tsndr/cloudflare-worker-jwt](https://github.com/tsndr/cloudflare-worker-jwt)
